src/http/server.cr

```diff
@@ -476,11 +480,17 @@ class HTTP::Server
     end
 
     @sockets.size.times { done.receive }
+
+    client_connection_count_dec
+    @stop_chan.receive?
   end
 
   # Gracefully terminates the server. It will process currently accepted
   # requests, but it won't accept new connections.
-  def close
+  #
+  # If *timeout* is specified #listen will stop waiting for client connections to close after *timeout* seconds
+  # Otherwise it waits for all connections to close.
```

Comment:
I think this documentation should be more explicit that there is no hard shutdown and you can either call `#close` with timeout and have zombie fibers hanging around, or call `#close` without timeout and `#listen` might never return.